helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -683,16 +692,11 @@ public Object call() {
     public void queueEvent(NotificationContext context, ClusterEventType eventType,
         HelixConstants.ChangeType changeType) {
       ClusterEvent event = new ClusterEvent(_clusterName, eventType);
-      if (context == null || context.getType() != NotificationContext.Type.CALLBACK
-          || context.getType() == NotificationContext.Type.PERIODIC_REFRESH) {
-        _dataCache.requireFullRefresh();
-      } else {
-        _dataCache.notifyDataChange(changeType, context.getPathChanged());
-      }
 
       // Null check for manager in the following line is done in handleEvent()
       event.addAttribute(AttributeName.helixmanager.name(), context.getManager());
       event.addAttribute(AttributeName.changeContext.name(), context);
+      event.addAttribute(AttributeName.ChangeType.name(), changeType);
```

Comment:
The type should be included in the context.

Reply:
Would changeType always be the same with context.getChangeType()? Is changeType always set in context? If answers are yes, we can remove this and use context.getChangeType().